core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -5592,6 +5592,63 @@ class ReplicaManagerTest {
     }
   }
 
+  @ParameterizedTest
+  @ValueSource(booleans = Array(true, false))
+  def testDeltaFollowerRecoverAbandonedFutureReplica(enableRemoteStorage: Boolean): Unit = {
+    // Given
+    val localId = 1
+    val topicPartition = new TopicPartition("foo", 0)
+
+    val mockReplicaFetcherManager = mock(classOf[ReplicaFetcherManager])
+    val mockReplicaAlterLogDirsManager = mock(classOf[ReplicaAlterLogDirsManager])
+    val replicaManager = setupReplicaManagerWithMockedPurgatories(
+      timer = new MockTimer(time),
+      brokerId = localId,
+      mockReplicaFetcherManager = Some(mockReplicaFetcherManager),
+      mockReplicaAlterLogDirsManager = Some(mockReplicaAlterLogDirsManager),
+      enableRemoteStorage = enableRemoteStorage,
+      shouldMockLog = true
+    )
+
+    val directoryId1 = Uuid.randomUuid()
+    val directoryId2 = Uuid.randomUuid()
+    val mockFutureLog = setupMockLog("/KAFKA-16082-test", isFuture = true)
+
+    val mockLogMgr = replicaManager.logManager
+    when(
+      mockLogMgr.getLog(topicPartition, isFuture = true)
+    ).thenReturn {
```

Comment:
`{` is used to hold a block but in this case it is a simple value. So maybe replace all `.thenReturn {Some(mockFutureLog)}` to `.thenReturn(Some(mockFutureLog))`